### ISSUE TITLE
chromeIPass is slow (adds hundreds of milliseconds to any page load time)

### ISSUE THREAD

**Issue by mehrdadn** [pfn/passifox#550]
1. Start a browUse the Chrome debugger's "Timeline" tab to see

**Comment by pfn** [comment]
I would appreciate a pr that would help address this.

**Comment by mehrdadn** [comment]
(I'll put that on my to-do list, but it'll take me quite a while to find the time and figure it out since I'm completely unfamiliar with the source code, so if someone gets the chance in the meantime that would be great.)

**Comment by mehrdadn** [comment]
OK I found one culprit. It's the `"all_frames": true` line in `manifest.json`.
The problem is that every single frame load causes jQuery to be loaded, which takes around 60 ms on my machine. For a page like Gmail, this adds > 400 ms to the page load time.

Any chance it could be changed to `false`? Or, even better, load jQuery only where it is necessary?

**Comment by pfn** [comment]
Actually, I would like to drop jquery altogether, but it's used relatively
pervasively (the toolbar icon dropdown utilizes jquery)

--

**Comment by mehrdadn** [comment]
Right, but in the absence of that, could you just change the flag to false? Or at least provide the option to do so on the settings page?

**Comment by pfn** [comment]
The problem is, if it's not loaded, frames that contain login forms will
not be processed.

**Comment by mehrdadn** [comment]
But that's why I said at least provide it as an advanced option? It's not an everyday thing for all users, so those of us who never come across it won't mind.

**Comment by pfn** [comment]
Manifest options cannot be changed at runtime

**Comment by mehrdadn** [comment]
Oh, right. Can jQuery loading be made dynamic instead? So that if the option is set, then jQuery is only loaded for the top level?

**Comment by pfn** [comment]
That would be something someone would make a PR for

**Comment by Hexstream** [comment]
Honestly loading chromeIPass on every single resource ever is just crazy, I'm also wasting 200+ms on every page load, even for completely unrelated resources like when I open an image directly.

This kind of stupid shit is partly why I dumped LastPass, though KeePassXC/ChromeIPass is certainly much better and I'll probably stick with it into the foreseeable future.

**Please fix this. This should be considered a high-priority issue.**
(And don't give me the "fix it yourself" crap. ;P)

**Comment by mehrdadn** [comment]
Interestingly I actually ended up "fixing" this myself in the code, the only trouble is it's totally breaking things I don't care about, but I can't really merge it back because of that...
(Edit: This is no longer the case.)

**Comment by mehrdadn** [comment]
@pfn Just wondering if you've seen my pull request for resolving this?

**Comment by ForsakenHarmony** [comment]
@pfn would you consider adding more contributors given that you seem to be busy and this extension is kinda stuck?
